src/hb-serialize.hh

```diff
@@ -127,7 +129,7 @@ struct hb_serialize_context_t
 
   template <typename T1, typename T2>
   bool check_assign (T1 &v1, T2 &&v2)
-  { return check_equal (v1 = v2, v2); }
+  { return check_equal ((hb_remove_reference<T2>) (v1 = v2), v2); }
```

Comment:
BUG:
If T1 = int32_t and T2 = unsigned = uint32_t (i.e., on 32-bit platform), this code fails to catch an integer overflow with v2 >= (1<<31).
Need a fix.

Reply:
Currently T2 is always `unsigned`, needs not a template argument, so casting to `uint64_t` should work as long as sizeof(T1) < sizeof(uint64_t):
```
template <typename T>
bool check_assign (T &v1, unsigned &&v2)
{ return check_equal ((uint64_t) (v1 = v2), v2); }

```
Better idea?

Reply:
Above code doesn't compile... trying to come up with a right fix.

Reply:
I pushed this, which should do?
https://github.com/harfbuzz/harfbuzz/commit/d663e28af1b84f27bea9730a9a8f18212387bb7f

Reply:
yes thanks